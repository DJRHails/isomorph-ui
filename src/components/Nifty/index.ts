export type { NiftyProps } from './Nifty';
export { Nifty } from './Nifty';