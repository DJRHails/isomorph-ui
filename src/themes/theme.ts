// Reference: https://github.com/ConsenSys/rimble-ui/blob/master/src/theme.js
// Reference: https://github.com/styled-system/styled-system/blob/master/docs/src/gatsby-plugin-theme-ui/index.js
// Reference Tailwind: https://github.com/system-ui/theme-ui/blob/76207a732fba4da145abf9535f33704b8811546c/packages/preset-tailwind/src/index.ts

import { Theme } from "styled-system";

export const shadows = {
  xs: "0 0 0 1px rgba(0, 0, 0, 0.05)",
  sm: "0 1px 2px 0 rgba(0, 0, 0, 0.05)",
  default: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
  md: "0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)",
  lg: "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
  xl:
    "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
  "2xl": "0 25px 50px -12px rgba(0, 0, 0, 0.25)",
  inner: "inset 0 2px 4px 0 rgba(0, 0, 0, 0.06)",
  outline: "0 0 0 2px var(--outline-color)",
  none: "none"
};

export const colors = {
  text: "#3f3d4b",
  surface: "#ffffff",
  brand: "#0A6EB4",
  interactive: "#0b77c2",
  interactiveHover: "#085387",
  interactiveText: "#ffffff",
  accent: "#f7b825",
  success: "#22543d",
  danger: "#dc3545",
  gray: "#82817d",
  muted: "#cccccc",
  badge: {
    brand: "#d7efff",
    danger: "#f5c6cb",
    success: "#c3e6cb",
  }
};

export const radii = {
  "none": "0",
  "sm": "0.125rem",
  "default": "0.25rem",
  "md": "0.375rem",
  "lg": "0.8rem",
  "full": "9999px",
};

export const space = [0, "0.25rem", "0.5rem", "1rem", "2rem", "4rem", "8rem"];

const tailwindSpacing = {
  px: "1px",
  "0": "0",
  "1": "0.25rem",
  "2": "0.5rem",
  "3": "0.75rem",
  "4": "1rem",
  "5": "1.25rem",
  "6": "1.5rem",
  "8": "2rem",
  "10": "2.5rem",
  "12": "3rem",
  "16": "4rem",
  "20": "5rem",
  "24": "6rem",
  "32": "8rem",
  "40": "10rem",
  "48": "12rem",
  "56": "14rem",
  "64": "16rem",
};

const tailwindMaxWidth = {
  xs: "20rem",
  sm: "24rem",
  md: "28rem",
  lg: "32rem",
  xl: "36rem",
  "2xl": "42rem",
  "3xl": "48rem",
  "4xl": "56rem",
  "5xl": "64rem",
  "6xl": "72rem",
};

const tailwindWidth = {
  "1/2": "50%",
  "1/3": "33.333333%",
  "2/3": "66.666667%",
  "1/4": "25%",
  "2/4": "50%",
  "3/4": "75%",
  "1/5": "20%",
  "2/5": "40%",
  "3/5": "60%",
  "4/5": "80%",
  "1/6": "16.666667%",
  "2/6": "33.333333%",
  "3/6": "50%",
  "4/6": "66.666667%",
  "5/6": "83.333333%",
  "1/12": "8.333333%",
  "2/12": "16.666667%",
  "3/12": "25%",
  "4/12": "33.333333%",
  "5/12": "41.666667%",
  "6/12": "50%",
  "7/12": "58.333333%",
  "8/12": "66.666667%",
  "9/12": "75%",
  "10/12": "83.333333%",
  "11/12": "91.666667%",
};

export const sizes = {
  ...tailwindSpacing,
  ...tailwindMaxWidth,
  ...tailwindWidth,
  full: "100%",
  screenHeight: "100vh",
  screenWidth: "100vw",
};

export const baseFonts = {
  sans:
    "\"Ubuntu\",system-ui,-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,\"Helvetica Neue\",Arial,\"Noto Sans\",sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\"",
  serif: "Georgia,Cambria,\"Times New Roman\",Times,serif",
  mono: "Menlo,Monaco,Consolas,\"Liberation Mono\",\"Courier New\",monospace"
};

export const fonts = {
  ...baseFonts,
  body: baseFonts.sans,
  heading: "inherit"
};

export const fontSizes = [
  "0.75rem",
  "1rem",    // p
  "1.25rem", // h5
  "1.50rem", // h4
  "2.25rem", // h3
  "3.50rem", // h2
  "5.00rem"  // h1
];

export const fontWeights = {
  light: 300,
  medium: 500,
  semibold: 600,
  bold: 700,
  black: 900,
};

export const lineHeights = {
  none: "1",
  tight: "1.25",
  snug: "1.375",
  normal: "1.5",
  relaxed: "1.625",
  loose: "2",
};

export const letterSpacings = {
  tighter: "-0.05em",
  tight: "-0.025em",
  normal: "0",
  wide: "0.025em",
  wider: "0.05em",
  widest: "0.1em"
};

export type TypographyVariants = "heading" | "body" | "button";
export const typography = {
  heading: {
    fontFamily: "body",
    color: "text",
    fontWeight: "bold",
    lineHeight: "none",
    m: 0,
    mb: 1,
  },
  body: {
    fontFamily: "body",
    fontSize: 1,
  },
  button: {
    position: "relative",
    fontSize: 1,
    fontWeight: "bold",
    zIndex: 1,
    userSelect: "none",
    pointerEvents: "none",
  }
};

const baseCard = {
  p: 3,
  borderRadius: "md",
  bg: "surface",
};

export type CardVariants = "outline" | "default";
const cards = {
  outline: {
    ...baseCard,
    "--outline-color": "rgb(226, 232, 240)",
    boxShadow: "outline",
  },
  default: {
    ...baseCard,
    boxShadow: "md",
  }
};

export type Variants = "badge" | "link";
const variants = {
  badge: {
    fontFamily: "body",
    textTransform: "uppercase",
    px: 2,
    py: 1,
    fontWeight: "bold",
    borderRadius: "sm",
    whiteSpace: "nowrap",
    display: "inline-block"
  },
  link: {
    color: "brand"
  }
};

export type VariantTypes = "variants" | "cards" | "typography" | "buttons";

// https://theme-ui.com/theme-spec/
export interface CoreTheme extends Theme {
  typography: typeof typography;
  cards: typeof cards;
  variants: typeof variants;
}
const theme: CoreTheme = {
  shadows,
  fontSizes,
  fontWeights,
  letterSpacings,
  lineHeights,
  fonts,
  colors,
  radii,
  space,
  sizes,

  // TODO: Change Variants to recommended system
  typography,
  cards,
  variants
};

export default theme;
