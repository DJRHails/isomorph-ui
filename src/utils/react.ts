/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types */
import React from 'react'

export function getValidChildren(children: React.ReactNode): React.ReactElement[] {
  return React.Children.toArray(children).filter((child) => 
    React.isValidElement(child),
  ) as React.ReactElement[]
}