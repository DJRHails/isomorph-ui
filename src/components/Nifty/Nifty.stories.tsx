import React, { ReactNode } from 'react';
import { KindMeta, StoryMeta, StoryFnType } from '/types/storybook';

import { Nifty, NiftyMetadata } from './Nifty';
import { Flex, Button } from '/components';

const withCenterComponent: StoryFnType<any, ReactNode> = (context) => <Flex
  width="full"
  height="screenHeight"
  bg="surface"
  alignItems="center"
>
  {context?.()}
</Flex>;

const story: KindMeta<typeof Nifty> = {
  title: 'Components/Nifty',
  component: Nifty,
  decorators: [withCenterComponent],
};

export const ARRtistryNiftyMetadata: NiftyMetadata = {
  src: 'internal://',
  assetName: 'Artifact',
  assetId: 16,
  owner: '0x67C3fD74fd36456FC5b339Fd09Efa330ad3126eE',

  // Optional Asset metadata
  assetTicker: 'ART',

  // Optional NiftyMetadata
  name: 'Gorilla',
  img: 'https://ipfs.io/ipfs/QmYe9Ngo3UBss48ud5Ny7byde3yXE1GeoWGuHH4oeekza6',
  bg: 'surface'
};

export const standard: StoryMeta = () => (
  <Nifty
    id={16}
    data={ARRtistryNiftyMetadata}
    platform="eth"
  />
);

export const flippable: StoryMeta = () => {
  const [flipped, setFlipped] = React.useState(false);
  return (
    <Nifty
      id={16}
      data={ARRtistryNiftyMetadata}
      platform="eth"
      flipped={flipped || undefined}
    >
      <hr/>
      <Flex px={3} py={3}>
        <Button variant="gradient" onClick={() => setFlipped(!flipped)}>
          Transfer
        </Button>
      </Flex>
    </Nifty>

  );
};


export default story;