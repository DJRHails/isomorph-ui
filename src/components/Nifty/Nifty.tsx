import React, { FC } from 'react';
import { box, Box, BoxProps, Image, WithPlatform, AspectRatio, Heading, Flex } from '/components';
import styled from 'styled-components';
import { BoxShadowProps, boxShadow } from 'styled-system';
import { PlatformDefiniton } from '../Platform/Platform';

// TODO: Migrate address to external
export type Address = string;

export interface AssetMetadata {
  assetName: string;
  assetId: number | Address; // The asset identifier
  owner: Address;
  assetTicker?: string;
  creator?: Address;
  manager?: Address; // Only set on Algorand
  reserve?: Address; // Only set on Algorand
  freeze?: Address; // Only set on Algorand
  clawback?: Address; // Only set on Algorand
}

export interface NiftyMetadata extends AssetMetadata {
  src: string; // Source of Metadata e.g. IPFS hash
  img?: string;
  bg?: string;
  name?: string;
}

export interface NiftyProps extends BoxProps, BoxShadowProps {
  id: number;
  data: NiftyMetadata;
  platform: PlatformDefiniton;
  flipped?: true;
  children?: React.ReactNode;
}

const StyledNifty = styled.div<BoxProps & BoxShadowProps>`
  ${box}
  ${boxShadow}
`;

export const Nifty: FC<NiftyProps> = ({
  id,
  data,
  platform,
  flipped,
  children,
}: NiftyProps) => {
  return (
    <StyledNifty
      width="full"
      maxWidth="md"
      mx="auto"
      boxShadow="xl"
    >
      <Box fontSize="5rem">
        <WithPlatform allowOverlap flipped={flipped} platform={platform}>
          <AspectRatio maxWidth="full">
            {data.img 
              ? <Image
                height="full"
                bg={data.bg || 'surface'}
                src={data.img}
              />
              : <Box width="full" bg={data.bg || 'surface'}/>
            }
          </AspectRatio>
        </WithPlatform>
      </Box>
      <Box px={3} py={5}>
        <Heading.h4>{data.name}</Heading.h4>
        <Heading.h5 color="muted">{data.assetName} #{id}</Heading.h5>
      </Box>
      <hr/>
      <Flex>
        <Flex>
          
        </Flex>
      </Flex>

      {children}
    </StyledNifty>
  );
};
Nifty.displayName='Nifty';
