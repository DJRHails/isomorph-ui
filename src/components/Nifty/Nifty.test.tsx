import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '/themes/theme';

import { Nifty } from './Nifty';

const MINIMAL_NIFTY_PROPS = {
  id: 0,
  data: {
    src: 'internal://',
    assetName: 'Artifact',
    assetId: 16,
    owner: '0x67C3fD74fd36456FC5b339Fd09Efa330ad3126eE'
  },
  platform: 'eth' 
} as const;

describe('Nifty component sanity', () => {
  it('has name', () => {
    expect(Nifty.displayName).toBe('Nifty');
  });

  it('matches default snapshot', () => {
    const component = render(<Nifty {...MINIMAL_NIFTY_PROPS}/>);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Nifty {...MINIMAL_NIFTY_PROPS}/>
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});