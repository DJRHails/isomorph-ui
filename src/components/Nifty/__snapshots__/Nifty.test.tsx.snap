// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nifty component sanity matches default snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznWOq kgWrrk"
        width="full"
      >
        <div
          class="sc-AxiKw indorZ"
          font-size="5rem"
        >
          <div
            class="sc-AxiKw fPVsD"
            font-size="1em"
          >
            <div
              class="sc-AxiKw estSfk"
            >
              <div
                class="sc-AxiKw GJeuC"
                width="full"
              />
            </div>
            <div
              class="sc-AxiKw iYSnmW"
              font-size="0.6em"
            >
              <div
                class="sc-AxiKw bkDCjy"
              >
                <div
                  class="sc-AxiKw hSFVmZ"
                >
                  <svg
                    class="sc-fzokOt jqHSVp"
                    color="brand"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                      fill="currentColor"
                    />
                    <path
                      d="M12 15.243l4.5-2.662-4.5 6.343-4.5-6.342 4.5 2.662v-.001zm0-10.74l4.5 7.087-4.5 2.662-4.5-2.662L12 4.502z"
                      fill="#fff"
                    />
                  </svg>
                </div>
                <div
                  class="sc-AxiKw fSecxJ"
                >
                  <svg
                    class="sc-fzokOt ebVCVY"
                    color="accent"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                      fill="currentColor"
                    />
                    <path
                      d="M8.027 17.36l1.696-2.937 1.696-2.927 1.686-2.937.28-.466.124.466.517 1.934-.58 1.003-1.696 2.927-1.685 2.937h2.027l1.696-2.937.879-1.52.414 1.52.786 2.937h1.82l-.786-2.937-.786-2.927-.207-.755 1.262-2.182h-1.841l-.062-.218-.641-2.4-.083-.31h-1.769l-.041.063-1.655 2.865-1.696 2.937-1.686 2.927L6 17.36h2.027z"
                      fill="#fff"
                      stroke="null"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxiKw hrhjKn"
        >
          <h4
            class="sc-AxiKw sc-AxhUy BqwwM"
            font-family="body"
            font-size="3"
            font-weight="bold"
          />
          <h5
            class="sc-AxiKw sc-AxhUy ybPKJ"
            color="muted"
            font-family="body"
            font-size="2"
            font-weight="bold"
          >
            Artifact
             #
            0
          </h5>
        </div>
        <hr />
        <div
          class="sc-AxiKw sc-fzoyAV dikUeM"
        >
          <div
            class="sc-AxiKw sc-fzoyAV dikUeM"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznWOq kgWrrk"
      width="full"
    >
      <div
        class="sc-AxiKw indorZ"
        font-size="5rem"
      >
        <div
          class="sc-AxiKw fPVsD"
          font-size="1em"
        >
          <div
            class="sc-AxiKw estSfk"
          >
            <div
              class="sc-AxiKw GJeuC"
              width="full"
            />
          </div>
          <div
            class="sc-AxiKw iYSnmW"
            font-size="0.6em"
          >
            <div
              class="sc-AxiKw bkDCjy"
            >
              <div
                class="sc-AxiKw hSFVmZ"
              >
                <svg
                  class="sc-fzokOt jqHSVp"
                  color="brand"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                    fill="currentColor"
                  />
                  <path
                    d="M12 15.243l4.5-2.662-4.5 6.343-4.5-6.342 4.5 2.662v-.001zm0-10.74l4.5 7.087-4.5 2.662-4.5-2.662L12 4.502z"
                    fill="#fff"
                  />
                </svg>
              </div>
              <div
                class="sc-AxiKw fSecxJ"
              >
                <svg
                  class="sc-fzokOt ebVCVY"
                  color="accent"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                    fill="currentColor"
                  />
                  <path
                    d="M8.027 17.36l1.696-2.937 1.696-2.927 1.686-2.937.28-.466.124.466.517 1.934-.58 1.003-1.696 2.927-1.685 2.937h2.027l1.696-2.937.879-1.52.414 1.52.786 2.937h1.82l-.786-2.937-.786-2.927-.207-.755 1.262-2.182h-1.841l-.062-.218-.641-2.4-.083-.31h-1.769l-.041.063-1.655 2.865-1.696 2.937-1.686 2.927L6 17.36h2.027z"
                    fill="#fff"
                    stroke="null"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-AxiKw hrhjKn"
      >
        <h4
          class="sc-AxiKw sc-AxhUy BqwwM"
          font-family="body"
          font-size="3"
          font-weight="bold"
        />
        <h5
          class="sc-AxiKw sc-AxhUy ybPKJ"
          color="muted"
          font-family="body"
          font-size="2"
          font-weight="bold"
        >
          Artifact
           #
          0
        </h5>
      </div>
      <hr />
      <div
        class="sc-AxiKw sc-fzoyAV dikUeM"
      >
        <div
          class="sc-AxiKw sc-fzoyAV dikUeM"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Nifty component sanity matches themed snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznWOq hPaBWi"
        width="full"
      >
        <div
          class="sc-AxiKw indorZ"
          font-size="5rem"
        >
          <div
            class="sc-AxiKw fPVsD"
            font-size="1em"
          >
            <div
              class="sc-AxiKw fgLlBD"
            >
              <div
                class="sc-AxiKw kbBPPd"
                width="full"
              />
            </div>
            <div
              class="sc-AxiKw iYSnmW"
              font-size="0.6em"
            >
              <div
                class="sc-AxiKw bkDCjy"
              >
                <div
                  class="sc-AxiKw epjIEg"
                >
                  <svg
                    class="sc-fzokOt dSWyFL"
                    color="brand"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                      fill="currentColor"
                    />
                    <path
                      d="M12 15.243l4.5-2.662-4.5 6.343-4.5-6.342 4.5 2.662v-.001zm0-10.74l4.5 7.087-4.5 2.662-4.5-2.662L12 4.502z"
                      fill="#fff"
                    />
                  </svg>
                </div>
                <div
                  class="sc-AxiKw cMMGnE"
                >
                  <svg
                    class="sc-fzokOt gTwld"
                    color="accent"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                      fill="currentColor"
                    />
                    <path
                      d="M8.027 17.36l1.696-2.937 1.696-2.927 1.686-2.937.28-.466.124.466.517 1.934-.58 1.003-1.696 2.927-1.685 2.937h2.027l1.696-2.937.879-1.52.414 1.52.786 2.937h1.82l-.786-2.937-.786-2.927-.207-.755 1.262-2.182h-1.841l-.062-.218-.641-2.4-.083-.31h-1.769l-.041.063-1.655 2.865-1.696 2.937-1.686 2.927L6 17.36h2.027z"
                      fill="#fff"
                      stroke="null"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxiKw eLjLHj"
        >
          <h4
            class="sc-AxiKw sc-AxhUy cypOSp"
            font-family="body"
            font-size="3"
            font-weight="bold"
          />
          <h5
            class="sc-AxiKw sc-AxhUy jNFJzU"
            color="muted"
            font-family="body"
            font-size="2"
            font-weight="bold"
          >
            Artifact
             #
            0
          </h5>
        </div>
        <hr />
        <div
          class="sc-AxiKw sc-fzoyAV dikUeM"
        >
          <div
            class="sc-AxiKw sc-fzoyAV dikUeM"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznWOq hPaBWi"
      width="full"
    >
      <div
        class="sc-AxiKw indorZ"
        font-size="5rem"
      >
        <div
          class="sc-AxiKw fPVsD"
          font-size="1em"
        >
          <div
            class="sc-AxiKw fgLlBD"
          >
            <div
              class="sc-AxiKw kbBPPd"
              width="full"
            />
          </div>
          <div
            class="sc-AxiKw iYSnmW"
            font-size="0.6em"
          >
            <div
              class="sc-AxiKw bkDCjy"
            >
              <div
                class="sc-AxiKw epjIEg"
              >
                <svg
                  class="sc-fzokOt dSWyFL"
                  color="brand"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                    fill="currentColor"
                  />
                  <path
                    d="M12 15.243l4.5-2.662-4.5 6.343-4.5-6.342 4.5 2.662v-.001zm0-10.74l4.5 7.087-4.5 2.662-4.5-2.662L12 4.502z"
                    fill="#fff"
                  />
                </svg>
              </div>
              <div
                class="sc-AxiKw cMMGnE"
              >
                <svg
                  class="sc-fzokOt gTwld"
                  color="accent"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                    fill="currentColor"
                  />
                  <path
                    d="M8.027 17.36l1.696-2.937 1.696-2.927 1.686-2.937.28-.466.124.466.517 1.934-.58 1.003-1.696 2.927-1.685 2.937h2.027l1.696-2.937.879-1.52.414 1.52.786 2.937h1.82l-.786-2.937-.786-2.927-.207-.755 1.262-2.182h-1.841l-.062-.218-.641-2.4-.083-.31h-1.769l-.041.063-1.655 2.865-1.696 2.937-1.686 2.927L6 17.36h2.027z"
                    fill="#fff"
                    stroke="null"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-AxiKw eLjLHj"
      >
        <h4
          class="sc-AxiKw sc-AxhUy cypOSp"
          font-family="body"
          font-size="3"
          font-weight="bold"
        />
        <h5
          class="sc-AxiKw sc-AxhUy jNFJzU"
          color="muted"
          font-family="body"
          font-size="2"
          font-weight="bold"
        >
          Artifact
           #
          0
        </h5>
      </div>
      <hr />
      <div
        class="sc-AxiKw sc-fzoyAV dikUeM"
      >
        <div
          class="sc-AxiKw sc-fzoyAV dikUeM"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
